src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -452,5 +498,35 @@ private async Task<List<string>> ExecuteCommandAsync(IExecutionContext context,
             return outputs;
         }
 #endif
+
+        private async Task CreateContainerNetworkAsync(IExecutionContext executionContext, string network)
+        {
+            Trace.Entering();
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            int networkExitCode = await _dockerManger.DockerNetworkCreate(executionContext, network);
+            if (networkExitCode != 0)
+            {
+                throw new InvalidOperationException($"Docker network create failed with exit code {networkExitCode}");
+            }
+            // Expose docker network to env
+            executionContext.Variables.Set(Constants.Variables.Agent.ContainerNetwork, network);
```

Comment:
are we set this variable over and over again? :D

Reply:
Not any more, this is only called once in StartContainersAsync